File: 2021/day01-1.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	contenu, _ := os.ReadFile("day01-puzzle-input.txt")

	tableau := strings.Split(string(contenu), "\n")
	compteur := 0
	precedente, _ :=  strconv.Atoi(tableau[0])

	for _, nombre := range tableau[1:] {
		entier, _ := strconv.Atoi(nombre)
		if entier > precedente {
			compteur += 1
			fmt.Println(entier, " plus grand que ", precedente)
		}
		precedente = entier
	}

	fmt.Println(compteur)
}

File: 2023/day01-1.py
f = open("day01-puzzle-input.txt", "r")

total = 0

for ligne in f:
    i = 0
    j = len(ligne) - 1

    while not ligne[i].isnumeric():
        i += 1

    while not ligne[j].isnumeric():
        j -= 1

    nombre = ligne[i] + ligne[j]

    total += int(nombre)

print(total)
